analyzers/src/SonarAnalyzer.CSharp/Rules/PropertyNamesShouldNotMatchGetMethods.cs

```diff
@@ -27,75 +27,58 @@
-                            additionalLocations: new[] { methodIdentifier.GetLocation() },
-                            messageArgs: new[] { propertyIdentifier.ValueText, methodIdentifier.ValueText }));
+                        c.ReportIssue(Diagnostic.Create(Rule, propertyIdentifier.GetLocation(), new[] { methodIdentifier.GetLocation() }, propertyIdentifier.ValueText, methodIdentifier.ValueText));
                     }
-                }, SyntaxKind.ClassDeclaration);
-        }
+                },
+                SyntaxKind.ClassDeclaration,
+                SyntaxKind.InterfaceDeclaration,
+                SyntaxKindEx.RecordClassDeclaration,
+                SyntaxKindEx.RecordStructDeclaration,
+                SyntaxKind.StructDeclaration);
 
-        private static IEnumerable<Tuple<SyntaxToken, SyntaxToken>> GetCollidingMembers(
-            IEnumerable<IPropertySymbol> properties, IEnumerable<IMethodSymbol> methods)
+        private static IEnumerable<Tuple<SyntaxToken, SyntaxToken>> CollidingMembers(IPropertySymbol[] properties, IMethodSymbol[] methods)
         {
             foreach (var property in properties)
             {
-                var collidingMethod = methods.FirstOrDefault(method => AreCollidingNames(property.Name, method.Name));
-                if (collidingMethod == null)
-                {
-                    continue;
-                }
-
-                var methodSyntax = collidingMethod.DeclaringSyntaxReferences.FirstOrDefault()?.GetSyntax()
-                    as MethodDeclarationSyntax;
-
-                if (!(property.DeclaringSyntaxReferences.FirstOrDefault()?.GetSyntax() is PropertyDeclarationSyntax propertySyntax) || methodSyntax == null)
+                if (methods.FirstOrDefault(x => AreCollidingNames(property.Name, x.Name)) is { } collidingMethod
+                    && collidingMethod.DeclaringSyntaxReferences.FirstOrDefault()?.GetSyntax() is MethodDeclarationSyntax methodSyntax
+                    && property.DeclaringSyntaxReferences.FirstOrDefault()?.GetSyntax() is PropertyDeclarationSyntax propertySyntax)
                 {
-                    continue;
+                    yield return new Tuple<SyntaxToken, SyntaxToken>(propertySyntax.Identifier, methodSyntax.Identifier);
                 }
-
-                yield return new Tuple<SyntaxToken, SyntaxToken>(propertySyntax.Identifier, methodSyntax.Identifier);
             }
         }
 
-        private static bool AreCollidingNames(string propertyName, string methodName)
-        {
-            return methodName.Equals(propertyName, StringComparison.OrdinalIgnoreCase) ||
-                methodName.Equals("Get" + propertyName, StringComparison.OrdinalIgnoreCase);
-        }
+        private static bool AreCollidingNames(string propertyName, string methodName) =>
+            methodName.Equals(propertyName, StringComparison.OrdinalIgnoreCase) || methodName.Equals("Get" + propertyName, StringComparison.OrdinalIgnoreCase);
```

Comment:
```suggestion
            methodName.Equals(propertyName, StringComparison.OrdinalIgnoreCase) || methodName.Equals($"Get{propertyName}", StringComparison.OrdinalIgnoreCase);
```

Reply:
I'll skip this one because I don't have any other changes to do here. Both options are equally long and produce identical IL.